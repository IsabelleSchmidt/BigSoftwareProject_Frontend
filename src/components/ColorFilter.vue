<template>
    <div class= "colorfilter">
        <h2>Farben</h2>
        <div class="colum1">
            <div class="onecolor">
                <input id="grey" class="inputfields" type="checkbox" value="grey" v-model="grey" @change="checked($event.target.value)"/>
                <label for="grey" class="grey"></label>
            </div>
            <div class="onecolor">
                <input id="beige" class="inputfields" type="checkbox" value="beige" v-model="beige" @change="checked($event.target.value)"/>
                <label for="beige" class="beige"></label>
            </div>
            <div class="onecolor">
                <input id="black" class="inputfields" type="checkbox" value="black" v-model="black" @change="checked($event.target.value)"/>
                <label for="black" class="black"></label>
            </div>
            <div class="onecolor">
                <input id="brown" class="inputfields" type="checkbox" value="brown" v-model="brown" @change="checked($event.target.value)"/>
                <label for="brown" class="brown"></label>
            </div>
        </div>
        <div class="column2">
            <div class="onecolor">
                <input id="green" class="inputfields" type="checkbox" value="green" v-model="green" @change="checked($event.target.value)"/>
                <label for="green" class="green"></label>
            </div>
            <div class="onecolor">
                <input id="white" class="inputfields" type="checkbox" value="white" v-model="white" @change="checked($event.target.value)"/>
                <label for="white" class="white"></label>
            </div>
            <div class="onecolor">
                <input id="blue" class="inputfields" type="checkbox" value="blue" v-model="blue" @change="checked($event.target.value)"/>
                <label for="blue" class="blue"></label>
            </div>
            <div class="onecolor">
                <input id="red" class="inputfields" type="checkbox" value="red" v-model="red" @change="checked($event.target.value)"/>
                <label for="red" class="red"></label>
            </div>
        </div>
        <div class="colum3">
            <div class="onecolor">
                <input id="yellow" class="inputfields" type="checkbox" value="yellow" v-model="yellow" @change="checked($event.target.value)"/>
                <label for="yellow" class="yellow"></label>
            </div>
            <div class="onecolor">
                <input id="orange" class="inputfields" type="checkbox" value="orange" v-model="orange" @change="checked($event.target.value)"/>
                <label for="orange" class="orange"></label>
            </div>
            <div class="onecolor">
                <input id="rosa" class="inputfields" type="checkbox" value="rosa" v-model="rosa" @change="checked($event.target.value)"/>
                <label for="rosa" class="rosa"></label>
            </div>
            <div class="onecolor">
                <input id="purple" class="inputfields" type="checkbox" value="purple" v-model="purple" @change="checked($event.target.value)"/>
                <label for="purple" class="purple"></label>
            </div>
        </div>  
    </div>  
</template>
<script lang="ts">
import { defineComponent } from 'vue'
export default defineComponent({
    name: "ColorFilter",

    setup(){
        function checked(color: string): void{
            console.log("color" + color)

        }
        return {
            checked
        };

    }
    
});
</script>
<style lang="scss">

.colorfilter{
    margin: 30px; 
}

label{
    cursor: pointer;
}
.inputfields{
    display: none;
}
.column1{
    float:left;
    width: 800px;
}
.column2{
    float: left;
    width:800px;
}
.column3{
    width: 800px;
    float: left;
}
.onecolor{
    float:left;
    margin: 15px; 
}
input[type=checkbox]:checked + label.grey::before {
  background: grey;
  color: white;
  content: "\2713";
  text-align: center;
  font-size: 36px;
  font-weight: bold;
}
input[type=checkbox]:checked + label.beige::before {
  background: beige;
  color: black;
  content: "\2713";
  text-align: center;
  font-size: 36px;
  font-weight: bold;
}
input[type=checkbox]:checked + label.black::before {
  background: black;
  color: white;
  content: "\2713";
  text-align: center;
  font-size: 36px;
  font-weight: bold;
}
input[type=checkbox]:checked + label.brown::before {
  background: brown;
  color: white;
  content: "\2713";
  text-align: center;
  font-size: 36px;
  font-weight: bold;
}
input[type=checkbox]:checked + label.green::before {
  background: green;
  color: white;
  content: "\2713";
  text-align: center;
  font-size: 36px;
  font-weight: bold;
}
input[type=checkbox]:checked + label.white::before {
  background: white;
  border: 1px solid black;
  color: #333;
  content: "\2713";
  text-align: center;
  font-size: 36px;
  font-weight: bold;
}
input[type=checkbox]:checked + label.blue::before {
  background: blue;
  color: white;
  content: "\2713";
  text-align: center;
  font-size: 36px;
  font-weight: bold;
}
input[type=checkbox]:checked + label.red::before {
  background: red;
  color: white;
  content: "\2713";
  text-align: center;
  font-size: 36px;
  font-weight: bold;
}
input[type=checkbox]:checked + label.yellow::before {
  background: yellow;
  color: black;
  content: "\2713";
  text-align: center;
  font-size: 36px;
  font-weight: bold;
}
input[type=checkbox]:checked + label.rosa::before {
  background: palevioletred;
  color: white;
  content: "\2713";
  text-align: center;
  font-size: 36px;
  font-weight: bold;
}
input[type=checkbox]:checked + label.purple::before {
  background: purple;
  color: white;
  content: "\2713";
  text-align: center;
  font-size: 36px;
  font-weight: bold;
}
input[type=checkbox]:checked + label.orange::before {
  background:orange;
  color: white;
  content: "\2713";
  text-align: center;
  font-size: 36px;
  font-weight: bold;
}

input[type=checkbox]+label.red::before{
    border-radius: 100px;
    background: red;
  content: "\00a0";
  display: inline-block;
  font: 16px/1em sans-serif;
  height: 35px;
  margin: 0 .25em 0 0;
  padding: 0;
  vertical-align: top;
  width: 50px;
}
input[type=checkbox]+label.grey::before{
    border-radius: 100px;
    background: grey;
  content: "\00a0";
  display: inline-block;
  font: 16px/1em sans-serif;
  height: 35px;
  margin: 0 .25em 0 0;
  padding: 0;
  vertical-align: top;
  width: 50px;
}
input[type=checkbox]+label.beige::before{
    border-radius: 100px;
    background:beige;
  content: "\00a0";
  display: inline-block;
  font: 16px/1em sans-serif;
  height: 35px;
  margin: 0 .25em 0 0;
  padding: 0;
  vertical-align: top;
  width: 50px;
}
input[type=checkbox]+label.black::before{
    border-radius: 100px;
    background: black;
  content: "\00a0";
  display: inline-block;
  font: 16px/1em sans-serif;
  height: 35px;
  margin: 0 .25em 0 0;
  padding: 0;
  vertical-align: top;
  width: 50px;
}
input[type=checkbox]+label.brown::before{
    border-radius: 100px;
    background: brown;
  content: "\00a0";
  display: inline-block;
  font: 16px/1em sans-serif;
  height: 35px;
  margin: 0 .25em 0 0;
  padding: 0;
  vertical-align: top;
  width: 50px;
}
input[type=checkbox]+label.green::before{
    border-radius: 100px;
    background:green;
  content: "\00a0";
  display: inline-block;
  font: 16px/1em sans-serif;
  height: 35px;
  margin: 0 .25em 0 0;
  padding: 0;
  vertical-align: top;
  width: 50px;
}
input[type=checkbox]+label.white::before{
    border-radius: 100px;
    border: 1px solid black;
    background: white;
  content: "\00a0";
  display: inline-block;
  font: 16px/1em sans-serif;
  height: 35px;
  margin: 0 .25em 0 0;
  padding: 0;
  vertical-align: top;
  width: 50px;
}
input[type=checkbox]+label.blue::before{
    border-radius: 100px;
    background:blue;
  content: "\00a0";
  display: inline-block;
  font: 16px/1em sans-serif;
  height: 35px;
  margin: 0 .25em 0 0;
  padding: 0;
  vertical-align: top;
  width: 50px;
}
input[type=checkbox]+label.yellow::before{
    border-radius: 100px;
    background:yellow;
  content: "\00a0";
  display: inline-block;
  font: 16px/1em sans-serif;
  height: 35px;
  margin: 0 .25em 0 0;
  padding: 0;
  vertical-align: top;
  width: 50px;
}
input[type=checkbox]+label.orange::before{
    border-radius: 100px;
    background:orange;
  content: "\00a0";
  display: inline-block;
  font: 16px/1em sans-serif;
  height: 35px;
  margin: 0 .25em 0 0;
  padding: 0;
  vertical-align: top;
  width: 50px;
}
input[type=checkbox]+label.rosa::before{
    border-radius: 100px;
    background: palevioletred;
  content: "\00a0";
  display: inline-block;
  font: 16px/1em sans-serif;
  height: 35px;
  margin: 0 .25em 0 0;
  padding: 0;
  vertical-align: top;
  width: 50px;
}
input[type=checkbox]+label.purple::before{
    border-radius: 100px;
    background:purple;
  content: "\00a0";
  display: inline-block;
  font: 16px/1em sans-serif;
  height: 35px;
  margin: 0 .25em 0 0;
  padding: 0;
  vertical-align: top;
  width: 50px;
}



</style>