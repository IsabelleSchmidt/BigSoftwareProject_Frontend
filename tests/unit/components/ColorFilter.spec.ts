jest.mock('checked', () => ({
    get: jest.fn()
  }))

import { shallowMount } from "@vue/test-utils"

import ColorFilter from "@/components/ColorFilter.vue"

describe('Colorfilter test', () => {
    
    const cmp = shallowMount(ColorFilter)
    
    
    test('trigger demo', async () => {
        
        const checked = cmp.find('checked')
        checked.element.textContent = "grey"
        checked.trigger('change')

  
        await cmp.trigger('change')

        expect(checked).toBeCalledWith('grey')
    })

    // test('if checked exists and is called', () => {
        
    //     expect(checked).toBeCalledWith('grey')

    // })

    // test('negative test --> is not allowed to trigger with the wrong input', () => {
    //     checked.trigger('checked')
    //     expect(cmp.emitted().checked).toBeFalsy()
    // })

    test('Snapshot match test ', () => {
        expect(cmp.html()).toMatchSnapshot()
        expect(cmp).toMatchSnapshot()
    })
})