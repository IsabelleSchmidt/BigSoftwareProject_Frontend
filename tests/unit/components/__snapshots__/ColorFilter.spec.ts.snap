// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colorfilter test Snapshot match test  1`] = `
<div class="colorfilter">
  <h2>Farben</h2>
  <div class="colum1">
    <div class="onecolor"><input id="grey" class="inputfields" type="checkbox" value="grau"><label for="grey" class="grey"></label></div>
    <div class="onecolor"><input id="beige" class="inputfields" type="checkbox" value="beige"><label for="beige" class="beige"></label></div>
    <div class="onecolor"><input id="black" class="inputfields" type="checkbox" value="schwarz"><label for="black" class="black"></label></div>
    <div class="onecolor"><input id="brown" class="inputfields" type="checkbox" value="braun"><label for="brown" class="brown"></label></div>
  </div>
  <div class="column2">
    <div class="onecolor"><input id="green" class="inputfields" type="checkbox" value="grün"><label for="green" class="green"></label></div>
    <div class="onecolor"><input id="white" class="inputfields" type="checkbox" value="weiss"><label for="white" class="white"></label></div>
    <div class="onecolor"><input id="blue" class="inputfields" type="checkbox" value="blau"><label for="blue" class="blue"></label></div>
    <div class="onecolor"><input id="red" class="inputfields" type="checkbox" value="rot"><label for="red" class="red"></label></div>
  </div>
  <div class="colum3">
    <div class="onecolor"><input id="yellow" class="inputfields" type="checkbox" value="gelb"><label for="yellow" class="yellow"></label></div>
    <div class="onecolor"><input id="orange" class="inputfields" type="checkbox" value="orange"><label for="orange" class="orange"></label></div>
    <div class="onecolor"><input id="rosa" class="inputfields" type="checkbox" value="rosa"><label for="rosa" class="rosa"></label></div>
    <div class="onecolor"><input id="purple" class="inputfields" type="checkbox" value="lila"><label for="purple" class="purple"></label></div>
  </div>
</div>
`;

exports[`Colorfilter test Snapshot match test  2`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "__emits": Object {},
      "__props": Array [
        Object {},
        Array [],
      ],
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <div
        class="colorfilter"
      >
        <h2>
          Farben
        </h2>
        <div
          class="colum1"
        >
          <div
            class="onecolor"
          >
            <input
              class="inputfields"
              id="grey"
              type="checkbox"
              value="grau"
            />
            <label
              class="grey"
              for="grey"
            />
          </div>
          <div
            class="onecolor"
          >
            <input
              class="inputfields"
              id="beige"
              type="checkbox"
              value="beige"
            />
            <label
              class="beige"
              for="beige"
            />
          </div>
          <div
            class="onecolor"
          >
            <input
              class="inputfields"
              id="black"
              type="checkbox"
              value="schwarz"
            />
            <label
              class="black"
              for="black"
            />
          </div>
          <div
            class="onecolor"
          >
            <input
              class="inputfields"
              id="brown"
              type="checkbox"
              value="braun"
            />
            <label
              class="brown"
              for="brown"
            />
          </div>
        </div>
        <div
          class="column2"
        >
          <div
            class="onecolor"
          >
            <input
              class="inputfields"
              id="green"
              type="checkbox"
              value="grün"
            />
            <label
              class="green"
              for="green"
            />
          </div>
          <div
            class="onecolor"
          >
            <input
              class="inputfields"
              id="white"
              type="checkbox"
              value="weiss"
            />
            <label
              class="white"
              for="white"
            />
          </div>
          <div
            class="onecolor"
          >
            <input
              class="inputfields"
              id="blue"
              type="checkbox"
              value="blau"
            />
            <label
              class="blue"
              for="blue"
            />
          </div>
          <div
            class="onecolor"
          >
            <input
              class="inputfields"
              id="red"
              type="checkbox"
              value="rot"
            />
            <label
              class="red"
              for="red"
            />
          </div>
        </div>
        <div
          class="colum3"
        >
          <div
            class="onecolor"
          >
            <input
              class="inputfields"
              id="yellow"
              type="checkbox"
              value="gelb"
            />
            <label
              class="yellow"
              for="yellow"
            />
          </div>
          <div
            class="onecolor"
          >
            <input
              class="inputfields"
              id="orange"
              type="checkbox"
              value="orange"
            />
            <label
              class="orange"
              for="orange"
            />
          </div>
          <div
            class="onecolor"
          >
            <input
              class="inputfields"
              id="rosa"
              type="checkbox"
              value="rosa"
            />
            <label
              class="rosa"
              for="rosa"
            />
          </div>
          <div
            class="onecolor"
          >
            <input
              class="inputfields"
              id="purple"
              type="checkbox"
              value="lila"
            />
            <label
              class="purple"
              for="purple"
            />
          </div>
        </div>
      </div>
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {
        "transition": Object {
          "name": "transition",
          "props": Object {},
          "render": [Function],
        },
        "transition-group": Object {
          "name": "transition-group",
          "props": Object {},
          "render": [Function],
        },
      },
      "config": Object {
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isCustomElement": [Function],
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "mixins": Array [
        Object {
          "__emits": null,
          "__props": Array [],
          "beforeCreate": [Function],
        },
        Object {
          "__emits": null,
          "__props": Array [],
          "beforeCreate": [Function],
        },
      ],
      "provides": Object {},
      "reload": [Function],
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": Object {
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isCustomElement": [Function],
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.0.4",
  },
  "__setProps": [Function],
  "componentVM": Object {
    "__emitted": Object {},
    "beige": undefined,
    "black": undefined,
    "blue": undefined,
    "brown": undefined,
    "checked": [Function],
    "green": undefined,
    "grey": undefined,
    "orange": undefined,
    "purple": undefined,
    "red": undefined,
    "rosa": undefined,
    "white": undefined,
    "yellow": undefined,
  },
  "rootVM": Object {
    "__emitted": Object {},
  },
}
`;
